app/assets/javascripts/reducers/assignments.js

```diff
@@ -35,6 +42,10 @@ export default function assignments(state = initialState, action) {
     }
     case UPDATE_ASSIGNMENT: {
       const updatedAssignment = action.data.assignment;
+      const assignmentExists = state.assignments.some(assignment => assignment.id === updatedAssignment.id);
```

Comment:
I think this may introduce a behavior change.

The UPDATE_ASSIGNMENT action is used for two things: "claiming" an assignment that is on the Available Articles list, and updating the sandbox url for an already-claimed assignment. The latter ought to have the same assignment ID, but the former may return a new different assignment ID. On the server, AssignmentManager uses the course's "retain available articles" setting to determine whether to update the existing assignment to associate it with a user (thereby moving it from the Available Articles list to the Assigned Articles list), or to create a duplicate to associated with the user while leaving the existing Assignment record unchanged (and still available for another user to choose from the Available Articles list).

So if the updatedAssignment is a newly-created duplicate, it will have an ID that is not already present in the state.

Reply:
Thank you for the clarification! I will update this and let you know